js/source/worker_tile.js

```diff
@@ -94,7 +94,7 @@ class WorkerTile {
                 });
 
                 bucket.populate(features, options);
-                featureIndex.bucketLayerIDs[bucket.index] = family.map(getLayerId);
+                featureIndex.bucketLayerIDs[bucket.index] = family.map((l) => l.id);
```

Comment:
The benchmarks don't show any meaningful performance difference.

### Written as is

benchmark | master b9d9509 | simplify-bucket b75bbf4
--- | --- | ---
**buffer** | 919 ms  | 914 ms 

### Variant 1

```
function serializeBuckets(buckets, transferables) {
    return buckets
        .filter((b) => !b.isEmpty())
        .map((b) => b.serialize(transferables));
}
```

benchmark | master b9d9509 | simplify-bucket b75bbf4
--- | --- | ---
**buffer** | 906 ms  | 911 ms 


### Variant 2

```
function bucketNonEmpty(bucket) {
    return !bucket.isEmpty();
}

function serializeBuckets(buckets, transferables) {
    return buckets
        .filter(bucketNonEmpty)
        .map((b) => b.serialize(transferables));
}
```

benchmark | master b9d9509 | simplify-bucket b75bbf4
--- | --- | ---
**buffer** | 893 ms  | 900 ms 

You can see that all the alternatives are within the range of the control (master b9d9509) -- ~890 - 920ms.

Variant 1 does avoid repeating the `filter`/`map` pipeline, so I'm inclined to go with it.